components/proxy/src/main/java/com/hotels/styx/metrics/reporting/sets/OperatingSystemMetricSet.java

```diff
@@ -0,0 +1,66 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.metrics.reporting.sets;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import com.sun.management.OperatingSystemMXBean;
+import com.sun.management.UnixOperatingSystemMXBean;
+
+import java.lang.management.ManagementFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Creates a set of gauges that monitor metrics provided by the JVM regarding the OS.
+ */
+public final class OperatingSystemMetricSet implements MetricSet {
+    private final OperatingSystemMXBean bean;
+
+    public OperatingSystemMetricSet() {
+        this.bean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
```

Comment:
Can we declare this as  "? extends Metric"? I haven't checked MetricSet.

Reply:
It doesn't compile that way, I'm afraid.